### PR TITLE
Migrate assert.NoError to require.NoError

### PR DESCRIPTION
**Pull request by tvolk131**
## Change Description
Migrate instances of `assert.NoError` to `require.NoError`. There are several thousand instances of the latter version in the codebase, but only a handful of the former. This change makes the calls consistent by always using `require`.

### CODE REVIEW

**Review by yyforyongyu — COMMENTED** [review]
Pending a release note, otherwise looks good to me. Thanks for the PR!

**Review by yyforyongyu — APPROVED** [review]
LGTM, thanks! Also we could aim for `v1.15.1` for the release note since it's a small change.

**Comment by tvolk131** [comment]
> Also we could aim for `v1.15.1` for the release note since it's a small change.

Done!

File: docs/release-notes/release-notes-0.15.1.md
Comment (guggero) [inline docs/release-notes/release-notes-0.15.1.md:23]: This cleanup is already being done in https://github.com/lightningnetwork/lnd/pull/6685, which we'll merge soon.

**Review by guggero — APPROVED** [review]
LGTM, thanks for the cleanup!

Needs a rebase. Though I'd wait until we merge https://github.com/lightningnetwork/lnd/pull/6685 because that will probably require you to rebase again.

File: docs/release-notes/release-notes-0.15.1.md
Comment (tvolk131) [reply docs/release-notes/release-notes-0.15.1.md]: Gotcha! Reverted.

**Comment by tvolk131** [comment]
@guggero #6685 is merged and this PR is rebased. I believe we can merge this now!

**Pull request merged**
